src/utils/logger.hpp

```diff
@@ -1,25 +1,67 @@
 #pragma once
 
+#include <functional>
+#include <iostream>
+
 #include <fmt/printf.h>
 
-extern int (*nrnpy_pr_stdoe_callback)(int, char*);
+class Logger {
+  public:
+    void setCallback(std::function<int(int, const char*)> cb) {
+        callback = std::move(cb);
+    }
+
+    const std::function<int(int, const char*)>& getCallback() const {
+        return callback;
+    }
+
+    // Unconditionnaly print non depending of the level
+    template <typename... Args>
+    void print(const char* fmt, Args&&... args) const {
+        output(1, fmt, std::forward<Args>(args)...);
+    }
+
+    // Use the old printer because sometimes the format string come from user
+    template <typename... Args>
+    void printf(const char* fmt, Args&&... args) const {
```

Comment:
As there is only 2 occurrences of this, I will remove this and handle the problem locally

Reply:
done